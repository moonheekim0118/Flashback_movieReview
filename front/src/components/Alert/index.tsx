import React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';


// 알림창 
const Alert=()=>{
    const { message, showAlert } = useSelector((state)=>state.alert);
    // showAlert 상태일때만 띄워준다. 

    return(
        <>
            {showAlert&&
            <Container>
                {message}
            </Container>}
        </>
    );
}

const Container = styled.div`
    position:fixed;
    left:50%;
    bottom:10%;
    transform:translateX(-50%);

    padding: 25px 15px;
    
    background-color:#cc00cc;
    border-raidus:10px;
    color:#fff;
    font-size:1.2rem;
    box-shadow:0px 0px 11px -1px rgba(0,0,0,0.75);

    z-index:7000;
`;

export default Alert;