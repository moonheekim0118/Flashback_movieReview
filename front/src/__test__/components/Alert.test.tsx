import 'jsdom-global/register'; 
import React from "react";
import { useSelector } from 'react-redux';
import { mount } from 'enzyme';
import Alert from '../../components/Alert';

jest.mock('react-redux', ()=>({
    useSelector : jest.fn(),
}));


describe('<Alert/>',()=>{
    let container;
    beforeEach(()=>{
        const state = { message: "test", showAlert:true }; 
        useSelector.mockImplementation(()=>state);
        container = mount( <Alert/> )
    })

    it('should match correctly', ()=>{
        expect(container.html()).toMatchSnapshot();
    })
});