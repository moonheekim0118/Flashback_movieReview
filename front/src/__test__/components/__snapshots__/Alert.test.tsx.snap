// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert/> should match correctly 1`] = `"<div class=\\"Alert__Container-l9ph2f-0 lwAmd\\">test</div>"`;
